packages/tizen_app_manager/lib/app_running_context.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'src/app_manager.dart';
+
+// Enumeration for application current state
+enum AppState {
+  // Undefined state
+  undefined,
+
+  // The UI Application is running in the foreground.
+  foreground,
+
+  // The UI Application is running in the background.
+  background,
+
+  // The service Application is running.
+  service,
+
+  // The service Application is terminated.
+  terminated
+}
+
+class AppRunningContext {
+  late AppContext _context;
+
+  final String appId;
+
+  bool get isTerminated => !_context.isAppRunning();
+
+  String get packageId => _context.getPackageId();
+
+  int get processId => _context.getProcessId();
+
+  AppState get appState => _stateConvert(_context.getAppState());
+
+  AppRunningContext({required this.appId, int handleAddress = 0}) {
+    _context = AppContext(appId, handleAddress);
+  }
+
+  static AppRunningContext fromMap(dynamic map) {
+    final appId = map['appId'] as String? ?? '';
+    final handle = map['handle'] as int;
```

Comment:
According to the C++ code, both values cannot be null so it could be just
```dart
    final String appId = map['appId'] as String;
    final int handle = map['handle'] as int;
```